### PR TITLE
[REVIEW] Add `nvtx` annotations for `Series` and `Index`

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR adds some missing `nvtx` annotations for `Series` and `Index`.

### CODE REVIEW

**Comment by galipremsagar** [comment]
This is a follow-up PR to: https://github.com/rapidsai/cudf/pull/10256

File: python/cudf/cudf/core/index.py [hunk 1]
```diff
@@ -434,21 +459,34 @@ def to_pandas(self):
             name=self.name,
         )
 
-    @property
+    @property  # type: ignore
```
Comment (vyasr) [inline python/cudf/cudf/core/index.py:462]: Why does this require a type ignore? We're not applying it after annotate in this case.

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -6998,12 +6998,12 @@ def _drop_rows_by_labels(
             raise KeyError("One or more values not found in axis")
 
         key_df = cudf.DataFrame(index=labels)
-        if isinstance(obj, cudf.Series):
+        if isinstance(obj, cudf.DataFrame):
+            return obj.join(key_df, how="leftanti")
+        else:
```
Comment (vyasr) [inline python/cudf/cudf/core/frame.py:7003]: Why did this order flip? It seems out of scope for this PR. Is it to make this work for Index objects too?

**Review by vyasr — COMMENTED** [review]
Could we consider defining a convenience wrapper around annotate so that we don't have to continually replicate the parameters? Something like

```
def cudf_annotate(func):
    return annotate(message=func.__name__, color="green", domain="cudf_python")(func)
```

You could always extract `func.__module__` or something as well if you want the message to be something more discriminative than just the name.

File: python/cudf/cudf/core/frame.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/frame.py:7003]: Yeah somehow adding annotations for this method triggered a `mypy` error for this piece of code.

File: python/cudf/cudf/core/index.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/index.py]: yes, this was an unnecessary find-replace.

**Comment by galipremsagar** [comment]
> Could we consider defining a convenience wrapper around annotate so that we don't have to continually replicate the parameters? Something like
> 
> ```
> def cudf_annotate(func):
>     return annotate(message=func.__name__, color="green", domain="cudf_python")(func)
> ```
> 
> You could always extract `func.__module__` or something as well if you want the message to be something more discriminative than just the name.

Yeah, I think we can do something like this. I'll add a follow-up PR for it as I'm yet to add this coverage to some more APIs

**Comment by vyasr** [comment]
> Yeah, I think we can do something like this. I'll add a follow-up PR for it as I'm yet to add this coverage to some more APIs, does that sound okay to you?

I'd be fine with merging this PR as is, but I would like to see something like `cudf_annotate` added before you add any further annotations so that you can immediately use it in subsequent PRs.

**Review by vyasr — APPROVED** [review]

**Comment by galipremsagar** [comment]
> > Yeah, I think we can do something like this. I'll add a follow-up PR for it as I'm yet to add this coverage to some more APIs, does that sound okay to you?
> 
> I'd be fine with merging this PR as is, but I would like to see something like `cudf_annotate` added before you add any further annotations so that you can immediately use it in subsequent PRs.

Sure 👍
